onnxruntime/core/framework/utils.cc

```diff
@@ -518,6 +546,69 @@ common::Status ExecuteGraph(const SessionState& session_state,
   return status;
 }
 
+#ifdef ENABLE_TRAINING
+common::Status ExecutePartialGraph(const SessionState& session_state, FeedsFetchesManager& feeds_fetches_manager,
+                                   const std::vector<OrtValue>& feeds, std::vector<OrtValue>& fetches, const bool& terminate_flag,
+                                   const logging::Logger& logger, PartialGraphExecutionState& state) {
+
+  // finalize the copy info using the provided feeds and fetches. will update device_copy_checks in the background
+  FinalizeFeedFetchCopyInfo(feeds_fetches_manager, feeds, fetches);
+  PartialExecutor executor{state, terminate_flag};
+  const auto& feeds_fetches_info = feeds_fetches_manager.GetFeedsFetchesInfo();
+  const auto& device_copy_checks = feeds_fetches_manager.GetDeviceCopyChecks();
+
```

Comment:
Can we refactor ExecuteGraphImpl to extract out the IExecutor creation (pass it in as a param) so that we can simply re-use the existing logic?

Reply:
@mrry has requested we do any refactoring as a seperate PR to avoid touching any inference code paths in this PR. All this copy-paste was done intentionally :)